version: "3.5"
services:
  cache:
    image: eqalpha/keydb
    container_name: kaavapino-keydb
    command: keydb-server /etc/keydb/keydb.conf --server-threads 2
    volumes:
      - cache_kaavoitus_api_data:/data
    ports:
      - "6379:6379"
    restart: on-failure

  kaavoitus-api:
    build:
      context: "."
      dockerfile: Dockerfile.dev
    image: kaavoitus-api
    volumes:
      - .:/kaavoitus-api:rw,cached
      - ./mock-data:/kaavoitus-api/mock-data
      - ./django-db/db.sqlite3:/kaavoitus-api/db/db.sqlite3
    environment:
      - DEBUG=1
      - ALLOWED_HOSTS=localhost
      - FACTA_DB_MOCK_DATA_DIR=mock-data/
      - USE_JSON_READER=True
      - KAAVAPINO_API_URL=https://kaavapino-testing.ddns.net/v1
    ports:
      - "8000:8000"
    container_name: kaavoitus-api

volumes:
  cache_kaavoitus_api_data:

